nomenclature/region_mapping_models.py

```diff
@@ -0,0 +1,71 @@
+from pydantic import BaseModel, ValidationError, validator, root_validator
+from typing import Dict, Iterable, Optional, List, Union
+from pathlib import Path
+import yaml
+
+
+class NativeRegion(BaseModel):
+    name: str  # name of the region in the incoming data
+    rename: Optional[str]  # new name for the scenario explorer
+    # this new name can be automatically verfied against
+    # the regions defined in definitions/regions/regions.yaml
+
+
+class CommonRegion(BaseModel):
+    name: str  # can also be verfied against the defined regions
+    constituent_regions: List[NativeRegion]  # list of native regions
+
+
+class RegionAggregationMapping(BaseModel):
+    model: str  # if we have a list of allowed models we could also verify the model name here
+    # we can require that we have to have least either native_regions or
+    # aggregation_regions
+    native_regions: Optional[List[NativeRegion]]
+    common_regions: Optional[List[CommonRegion]]
+
+
+# should probably be moved inside RegionAggregationMapping as a classmethod
+def convert_region_mapping(file: Union[Path, str]) -> RegionAggregationMapping:
+    with open(file, "r") as stream:
+        mapping_input = yaml.safe_load(stream)
+    # check if we have a key called
+    # we need a model name and either native or common regions
+    if not "model" and ("native_regions" or "common_regions") in mapping_input.keys():
+        raise ValueError(
+            "Region mapping needs an attribute 'model' and at "
+            "least one of the following: 'native_regions',"
+            "'common_regions'"
+        )
+    if "native_regions" in mapping_input.keys():
```

Comment:
No need to specify `keys()`...
```suggestion
    if "native_regions" in mapping_input:
```